generator/internal/service_code_generator.cc

```diff
@@ -51,7 +57,30 @@ ServiceCodeGenerator::ServiceCodeGenerator(
       service_vars_(std::move(service_vars)),
       service_method_vars_(std::move(service_method_vars)),
       header_(context, service_vars_[header_path_key]) {
+  assert(service_descriptor != nullptr);
+  assert(context != nullptr);
   SetVars(service_vars_[header_path_key]);
+  for (int i = 0; i < service_descriptor_->method_count(); ++i) {
+    methods_.emplace_back(*service_descriptor_->method(i));
+  }
+}
+
+VarsDictionary const& ServiceCodeGenerator::vars() const {
+  return service_vars_;
+}
+
+std::string ServiceCodeGenerator::vars(std::string const& key) const {
```

Comment:
Nit: it seems like this method should probably be named `var` since it's returning the value for a single var. But I'll leave it up to you.